doc/modules/linear_model.rst

```diff
@@ -897,6 +897,169 @@ to warm-starting (see :term:`Glossary <warm_start>`).
+Poisson, Gamma and Inverse Gaussian distributions don't support negative values,
+it is convenient to apply a link function different from the identity link
+:math:`h(x^\top w)=x^\top w` that guarantees the non-negativeness, e.g. the
+log-link `link='log'` with :math:`h(x^\top w)=\exp(x^\top w)`.
+
+:class:`TweedieRegressor` implements a generalized linear model
+for the Tweedie distribution, that allows to model any of the above mentioned
+distributions using the appropriate ``power`` parameter, i.e. the exponent
+of the unit variance function:
+
+ - ``power = 0``: Normal distribution. Specialized solvers such as
+   :class:`Ridge`, :class:`ElasticNet` are generally
+   more appropriate in this case.
+
+ - ``power = 1``: Poisson distribution. :class:`PoissonRegressor` is exposed for
+   convenience. However, it is strictly equivalent to
+   `TweedieRegressor(power=1)`.
+
+ - ``power = 2``: Gamma distribution. :class:`GammaRegressor` is exposed for
+   convenience. However, it is strictly equivalent to
+   `TweedieRegressor(power=2)`.
+
+ - ``power = 3``: Inverse Gamma distribution.
+
+
+.. note::
+
+   * The feature matrix `X` should be standardized before fitting. This
+     ensures that the penalty treats features equally.
+   * If you want to model a relative frequency, i.e. counts per exposure (time,
+     volume, ...) you can do so by a Poisson distribution and passing
+     :math:`y=\frac{\mathrm{counts}}{\mathrm{exposure}}` as target values
+     together with :math:`s=\mathrm{exposure}` as sample weights. This is done
+     in both examples linked below.
+   * The fit itself does not need Y to be from an EDM, but only assumes
+     the first two moments to be :math:`E[Y_i]=\mu_i=h((Xw)_i)` and
+     :math:`Var[Y_i]=\frac{\phi}{s_i} v(\mu_i)`.
+   * If the target `y` is a ratio, appropriate sample weights ``s`` should be
+     provided.
+     As an example, consider Poisson distributed counts z (integers) and
+     weights s=exposure (time, money, persons years, ...). Then you fit
+     y = z/s, i.e. ``PoissonRegressor.fit(X, y, sample_weight=s)``.
+     The weights are necessary for the right (finite sample) mean.
+     Considering :math:`\bar{y} = \frac{\\sum_i s_i y_i}{\sum_i s_i}`,
+     in this case one might say that y has a 'scaled' Poisson distribution.
+     The same holds for other distributions.
+
+The estimator can be used as follows::
+
+    >>> from sklearn.linear_model import TweedieRegressor
+    >>> reg = TweedieRegressor(power=1, alpha=0.5, link='log')
+    >>> reg.fit([[0, 0], [0, 1], [2, 2]], [0, 1, 2])
+    TweedieRegressor(alpha=0.5, link='log', power=1)
+    >>> reg.coef_
+    array([0.2463..., 0.4337...])
+    >>> reg.intercept_
+    -0.7638...
+
+
+.. topic:: Examples:
+
+  * :ref:`sphx_glr_auto_examples_linear_model_plot_tweedie_regression_insurance_claims.py`
+  * :ref:`sphx_glr_auto_examples_linear_model_plot_poisson_regression_non_normal_loss.py`
+
+Mathematical formulation
+------------------------
```

Comment:
I'm not really convinced that we need this section. It seems to me that it is either redundant with what's already above, or too specific for a user-guide.

I think we can remove it?